### DISCUSSION TITLE
How to hide the waybar when there is a game in that workspace?

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [hyprwm/Hyprland#6972] [Q&A]
I want to hide the waybar where there is a game in that workspace. How can I do that? It's freezing when I  go fullscreen.
Game: AOE2DE

**Comment by ildar** [comment]
go fullscreen:
> bind = $mainMod, F, fullscreen, 1

**Reply by azizsayfiddin** [reply]
It's freezing when I go to fullscreen.

**Comment by Fanccy315** [comment]
I set the following key bindings

```
bind = SUPER,W,exec,killall -SIGUSR1 waybar
bind = SUPER SHIFT,W,exec,killall -SIGUSR2 waybar
```

So when I press the Win + W, waybar will hide
Win + Shift + W has the opposite function

**Reply by azizsayfiddin** [reply]
How can I make a single toggle bind, so that when I press SUPER+W, it hides the window if it's visible, and shows it if it's hidden?

**Reply by Fanccy315** [reply]
As far as I know, in the latest waybar git version, the first key binding just has the function you want.You can wait for the next release.
Besides, in the game, you can try enabling "full screen" instead of "borderless window"(if they exist).

**Reply by azizsayfiddin** [reply]
> As far as I know, in the latest waybar git version, the first key binding just has the function you want.You can wait for the next release. Besides, in the game, you can try enabling "full screen" instead of "borderless window"(if they exist).

The game (AoE2DE) freezes (stop rendering) when I go to fullscreen.

**Reply by Phantomwise** [reply]
You don't need a specific waybar function to make a toggle bind :
```
bind = $mainMod SHIFT, W, exec, killall waybar || waybar
```
If waybar is already running then it kills waybar, and if waybar is not running then it launches waybar.
I find it more useful than an option to hide/show waybar would be, because it actually kills the process so it can be used to restart waybar if it breaks or freezes, which is very handy.

Other stuff you might try :

Maybe the issue is not with fullscreen in general but with switching from a normal tiled window to fullscreen ? That happens with some games that don't play nice with tilers --though for me it's usually the reverse : some games launch fine in fullscreen on their own, but all hell breaks loose if I exit the fullscreen mode.

You could set a window rule to start this game automatically in fullscreen and see if it still freezes.

Or you could try to make the window floating first (I can't remember the default keybind), then fullscreen it. The game might do better transitioning from floating to fullscreen than it does from tiled to fullscreen. If that works, then you can make a window rule to make the game always launch in a floating window, so you don't have to manually do it every time.

For that you need to get the window class for the game by running `hyprctl clients` in a terminal while the game is open. The command will give you a list of every opened window and their details, look for the section for your game's window, and inside it you'll have `class: YourGame`

Then, add this to your config to make the game launch in fullsceen :
```
windowrulev2 = fullscreen, class:(kitty)
```
Or this to make it launch in a floating window :
```
windowrulev2 = float, class:(YourGame)
```
